PayPalNativeCheckout/src/main/java/com/paypal/android/checkout/PayPalClient.kt

```diff
@@ -14,36 +14,35 @@ import com.paypal.checkout.error.OnError
 /**
  * Use this client to checkout with PayPal.
  */
-class PayPalClient(application: Application, coreConfig: CoreConfig, returnUrl: String) {
+class PayPalClient internal constructor(
+    private val application: Application,
+    private val coreConfig: CoreConfig,
+    private val returnUrl: String,
+    private val api: API
+) {
+
+    constructor(application: Application, coreConfig: CoreConfig, returnUrl: String) :
+            this(application, coreConfig, returnUrl, API(coreConfig))
 
     /**
      * Sets a listener to receive notifications when a PayPal event occurs.
      */
     var listener: PayPalListener? = null
 
-    init {
-        val clientId = coreConfig.clientId
-        if (clientId.isNullOrEmpty()) {
-            throw PayPalSDKError(
-                0,
-                "Client Id should not be null or empty"
-            )
-        }
+    /**
+     * Initiate a PayPal checkout for an order.
+     *
+     * @param orderId the id of the order
+     */
+    suspend fun checkout(orderId: String) {
```

Comment:
wondering if we should just use a `CoroutineScope` inisde the `checkout` function. It does feel "weird" to use a suspend functions for what essentially is a UI flow from the merchants point of view. Thinks also about `java` integrations...

Reply:
Yeah that was honestly the path of least resistance. We were already calling checkout in a coroutine so it felt ok. We can also circle back once we start integrating NXO again.

Reply:
sure, its out of the scope of this PR, will get back to it on the nxo pr